node.c

```diff
@@ -0,0 +1,74 @@
+#include "node.h"
+#include "board.h"
+
+
+void initializeRoot(node* root, struct board *boardState){
+
+    // Setting default values
+    root->parent = NULL;
+    root->childMoves = (int *)malloc(2 * sizeof(int));
+    root->numChildren = 0;
+    root->numWins = 0;
+    root->board = *boardState;
+    root->numSimulations = 0;
+    root->children = (node **)malloc(sizeof(node*));
+}
+
+
+node* expand(node* parentNode, int* action, int name){
+
+    // Dynamically allocate new struct node
+    node* newNode = (node *)malloc(sizeof(node));
+
+    // Setting member variables for new node
+    newNode->parent = parentNode;
+    newNode->childMoves = (int *)malloc(2 * sizeof(int));
+    newNode->numChildren = 0;
+    newNode->numWins = 0;
+    newNode->numSimulations = 0;
+    newNode->children = (node **)malloc(sizeof(node*));
+
+    copyBoard(&parentNode->board, &newNode->board);
+    placePiece(&newNode->board, action);
+
+
+    // Update parent node
+    parentNode->children = (node **)realloc(parentNode->children, (parentNode->numChildren + 1) * sizeof(node*));
+    parentNode->children[parentNode->numChildren] = newNode;
+    parentNode->childMoves = (int *)realloc(parentNode->childMoves, (parentNode->numChildren + 1) * 2 * sizeof(int));
```

Comment:
Same as line 36, but you should quadruple the size of this array every time you fill it as it gets filled twice as fast as the children array. Again, you can check if the number of children is divisible by two to know when to resize this.

 - [ ] Update size of childMoves array by quadrupling the size intermittently rather than increasing the size by 2 every time